plugins/main/ipvlan/ipvlan.go

```diff
@@ -175,9 +189,20 @@ func cmdAdd(args *skel.CmdArgs) error {
 	var result *current.Result
 	// Configure iface from PrevResult if we have IPs and an IPAM
 	// block has not been configured
-	if n.IPAM.Type == "" && n.PrevResult != nil && len(n.PrevResult.IPs) > 0 {
-		result = n.PrevResult
-	} else {
+	haveResult := false
+	if n.IPAM.Type == "" && n.PrevResult != nil {
+		if err := version.ParsePrevResult(&n.NetConf); err != nil {
```

Comment:
Can't we just use the n.PrevResult here? That already go parsed in loadConf() so we should just be able to call NewResultFromResult() and check len(IPs)